### PR TITLE
`require-example` linting

### PR DESCRIPTION
**Pull request by karimnaaji**
Continuing on fixing linter warnings `require-example`.

### CODE REVIEW

File: src/source/image_source.js [hunk 1]
```diff
@@ -145,6 +145,28 @@ class ImageSource extends Evented implements Source {
...
+     *     ]
+     * });
+     * // Then update the image URL and coordinates
+     * imageSource.updateImage({
+     *     url: 'https://www.mapbox.com/images/bar.png',
+     *     coordinates: [
+     *         [-76.54335737228394, 39.18579907229748],
+     *         [-76.52803659439087, 39.1838364847587],
+     *         [-76.5295386314392, 39.17683392507606],
+     *         [-76.54520273208618, 39.17876344106642]
```
Comment (ryanhamley) [inline src/source/image_source.js:167]: nit: We should probably truncate lng/lat numbers

File: src/source/image_source.js [hunk 2]
```diff
@@ -175,6 +197,25 @@ class ImageSource extends Evented implements Source {
...
+     *         [-76.52, 39.17],
+     *         [-76.54, 39.17]
+     *     ]
+     * });
+     * // Then update the image coordinates
+     * imageSource.setCoordinates([
+     *     [-76.54335737228394, 39.18579907229748],
+     *     [-76.52803659439087, 39.1838364847587],
+     *     [-76.5295386314392, 39.17683392507606],
+     *     [-76.54520273208618, 39.17876344106642]
```
Comment (ryanhamley) [inline src/source/image_source.js:217]: here too

File: src/source/vector_tile_source.js [hunk 3]
```diff
@@ -155,6 +155,18 @@ class VectorTileSource extends Evented implements Source {
      *
      * @param {string[]} tiles An array of one or more tile source URLs, as in the TileJSON spec.
      * @returns {VectorTileSource} this
+     * @example
+     * map.addSource('vector_source_id', {
+     *     type: 'vector',
+     *     tiles: ['https://d25uarhxywzl1j.cloudfront.net/v0.1/{z}/{x}/{y}.mvt'],
```
Comment (ryanhamley) [inline src/source/vector_tile_source.js:161]: Where did this URL come from? Is this something we control? Maybe it'd be better to use a fake URL.

File: src/source/vector_tile_source.js [hunk 4]
```diff
@@ -169,6 +181,16 @@ class VectorTileSource extends Evented implements Source {
      *
      * @param {string} url A URL to a TileJSON resource. Supported protocols are `http:`, `https:`, and `mapbox://<Tileset ID>`.
      * @returns {VectorTileSource} this
+     * @example
+     * map.addSource('vector_source_id', {
+     *     type: 'vector',
+     *     url: 'mapbox://mapbox.mapbox-streets-v6'
```
Comment (ryanhamley) [inline src/source/vector_tile_source.js:187]: nit: let's use the most up-to-date sources which are v7 and v8

File: src/source/video_source.js [hunk 5]
```diff
@@ -151,6 +168,30 @@ class VideoSource extends ImageSource {
...
+     *     ]
+     * });
+     *
+     * // Then update the video source coordinates by new coordinates
+     * const videoSource = map.getSource('video_source_id');
+     * videoSource.setCoordinates([
+     *     [-76.54335737228394, 39.18579907229748],
+     *     [-76.52803659439087, 39.1838364847587],
+     *     [-76.5295386314392, 39.17683392507606],
+     *     [-76.54520273208618, 39.17876344106642]
```
Comment (ryanhamley) [inline src/source/video_source.js:193]: coords

File: src/ui/camera.js [hunk 6]
```diff
@@ -950,6 +972,16 @@ class Camera extends Evented {
...
+     * @example
+     * const camera = map.getFreeCameraOptions();
+     *
+     * const position = [138.72649, 35.33974];
+     * const altitude = 3000;
+     *
+     * camera.position = mapboxgl.MercatorCoordinate.fromLngLat(position, altitude);
+     * camera.lookAtPoint([138.73036, 35.36197]);
+     *
+     * map.setFreeCameraOptions(camera);
```
Comment (ryanhamley) [inline src/ui/camera.js:984]: this example should be for `getFreeCameraOptions` it looks like

File: src/ui/free_camera.js [hunk 7]
```diff
@@ -126,6 +136,16 @@ class FreeCameraOptions {
...
+     * @example
+     * const camera = map.getFreeCameraOptions();
+     *
+     * const position = [138.72649, 35.33974];
+     * const altitude = 3000;
+     *
+     * camera.position = mapboxgl.MercatorCoordinate.fromLngLat(position, altitude);
+     * camera.lookAtPoint([138.73036, 35.36197]);
+     *
+     * map.setFreeCameraOptions(camera);
```
Comment (ryanhamley) [inline src/ui/free_camera.js:148]: This should be `lookAtPoint`

**Review by ryanhamley — CHANGES_REQUESTED** [review]
I had a couple nits but it also looks like a couple of the free camera examples need to be updated

File: src/ui/camera.js (hunk 6 above)
Comment (karimnaaji) [reply src/ui/camera.js:984]: I've chosen to be thorough there, explicitly making full use of the free camera API through getFreeCamera/update options/setFreeCamera, I think it's best to leave as is, so someone who has not been exposed to the entire API gets exposed to the full cycle.

Removing it means we end up having a `camera.lookAtPoint`, but where does the `camera` variable comes from? 

Best to leave as is IMO.

File: src/ui/free_camera.js (hunk 7 above)
Comment (karimnaaji) [reply src/ui/free_camera.js:148]: Same reasoning as above.

**Review by ryanhamley — APPROVED** [review]

**Pull request merged**